extensions/mailer/runtime/src/main/java/io/quarkus/mailer/runtime/MailerRecorder.java

```diff
@@ -0,0 +1,101 @@
+package io.quarkus.mailer.runtime;
+
+import java.util.function.Supplier;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.mailer.MailTemplate;
+import io.quarkus.mailer.Mailer;
+import io.quarkus.mailer.MailerName;
+import io.quarkus.mailer.MockMailbox;
+import io.quarkus.mailer.reactive.ReactiveMailer;
+import io.quarkus.qute.Template;
+import io.quarkus.qute.runtime.LocationLiteral;
+import io.quarkus.runtime.annotations.Recorder;
+import io.vertx.ext.mail.MailClient;
+
+@Recorder
+public class MailerRecorder {
```

Comment:
As an improvement we can make later on, we can use `SyntheticCreationalContext` instead of obtaining the beans from Arc manually

Reply:
I'll have a look at that today.

Reply:
Done, thanks for the hint!

Reply:
👌